multicluster/controllers/multicluster/core/remote_cluster_manager.go

```diff
@@ -0,0 +1,179 @@
+/*
+Copyright 2021 Antrea Authors.
+
+Licensed under the Apache License, Version 2.0 (the "License");
+you may not use this file except in compliance with the License.
+You may obtain a copy of the License at
+
+    http://www.apache.org/licenses/LICENSE-2.0
+
+Unless required by applicable law or agreed to in writing, software
+distributed under the License is distributed on an "AS IS" BASIS,
+WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+See the License for the specific language governing permissions and
+limitations under the License.
+*/
+
+package core
+
+import (
+	"context"
+	"sync"
+
+	"k8s.io/klog/v2"
+
+	"antrea.io/antrea/multicluster/controllers/multicluster/common"
+)
+
+type clusterEvent struct {
+	isAdd         bool
+	remoteCluster RemoteCluster
+}
+
+type RemoteClusterManager interface {
+	// Start RemoteClusterManager on an event loop in a go routine
+	Start() error
+	// Stop RemoteClusterManager
+	Stop() error
+	// AddRemoteCluster adds a remote cluster to RemoteClusterManager.
+	AddRemoteCluster(remoteCluster RemoteCluster)
+	// RemoveRemoteCLuster removes a remote cluster from RemoteClusterManager.
+	RemoveRemoteCluster(remoteCluster RemoteCluster)
+	// GetRemteClusters returns all remote clusters
+	GetRemoteClusters() map[common.ClusterID]RemoteCluster
+	// GetElectedLeaderClusterID returns the leader remote cluster or INVALID_CLUSTER_ID if none elected
+	GetElectedLeaderClusterID() common.ClusterID
+	// GetLocalClusterID returns local cluster ID
+	GetLocalClusterID() common.ClusterID
+}
+
+// The manager manages all leaders defined in the cluster set and manages common area access into them.
+// It also interfaces with leader election to know the elected-leader from which resources
+// must be monitored.
+type remoteClusterManager struct {
+	// Synchronize access between leader election and changes
+	// to cluster list
+	mutex sync.Mutex
```

Comment:
this mutex is not used, I suppose you should call this in following code as r.mutex?

Reply:
It will be used in the unit-test patch, because I fixed some race conditions in that.